modules/java/native/java-destination.c

```diff
@@ -173,7 +173,15 @@ java_dd_deinit(LogPipe *s)
 gint
 java_dd_send_to_object(JavaDestDriver *self, LogMessage *msg)
 {
-  return java_destination_proxy_send(self->proxy, msg);
+  gint result = java_destination_proxy_send(self->proxy, msg);
+  if (result < 0 || result >= WORKER_INSERT_RESULT_MAX)
+    {
+      msg_error("java_destination: worker insert result out of range. Retrying message later",
```

Comment:
optional: the message could include destination location tag

Reply:
Good idea, fixed thanks.